### DISCUSSION TITLE
cel-go

### DISCUSSION THREAD

**Discussion by jackieli-tes** [hyperjumptech/grule-rule-engine#218] [Ideas]
I see a lot of the the evaluation process can utilise a library like [cel-go](https://github.com/google/cel-go). Is this considered or decided against?

**Comment by newm4n** [comment]
The GRL used in Grule is already tailored as a complete DSL for simple rule engine. I dont see a reason on why we should use a more generic single expression evaluation engine. Using cel-go will give us hard time to index expressions and expression atoms for our RETE algorithm, as expressions are handled by the library, not by grule.

So, I can say that its already considered and decided nit to use one.
